onebusaway-android/src/main/java/org/onebusaway/android/tripservice/TripService.java

```diff
@@ -111,33 +111,34 @@ public TaskContextImpl(int startId) {
         public void taskComplete() {
             //Log.d(TAG, "Task complete: " + mStartId);
             // If we have notifications, then we can't stop ourselves.
-            if (mNotifications.isEmpty()) {
+            //TODO: Is there actually a reason to not stop ourselves if we are no longer storing notifications?
```

Comment:
Seems like the simplest solution without a complete overahul of the code (e.g.  to use something like an `IntentService`) may be to keep the `mNotifications` object, but only use it as a means to ensure we don't shut down the service prematurely.    `SetLatestEventInfo()` should still be eliminated.